streaming/src/main/scala/org/apache/spark/streaming/rdd/WriteAheadLogBackedBlockRDD.scala

```diff
@@ -136,7 +139,7 @@ class WriteAheadLogBackedBlockRDD[T: ClassTag](
         // this dummy directory should not already exist otherwise the WAL will try to recover
         // past events from the directory and throw errors.
         val nonExistentDirectory = new File(
-          System.getProperty("java.io.tmpdir"), UUID.randomUUID().toString).getAbsolutePath
+          tmpDir, UUID.randomUUID().toString).getAbsolutePath
```

Comment:
I don't think a new configuration is the right thing here.

Seems like the problem here is that this should really be pointing at the local disk, and not `fs.defaultFS`; in which case, instead of `getAbsolutePath`, it should be `.toURI().toString()`. Does that work?

Reply:
@vanzin I wondered the same thing I figured for now I'd generate the patch to try to get some feedback. This only affects the YARN Cluster mode as it looks $PWD which resolves to the local disk where the job is executing. I can look at making the change to use localFS if it seems reasonable? does that work?

Reply:
> does that work?

That's my question. I don't like a new configuration. So I'd only consider going down that path if there is no other alternative.

Reply:
I'll run some tests over here on my cluster with a few changes and see what happens

Reply:
Basically what I'm suggesting with my latest top-level comment, and what I first suggested here, is to replace `getAbsolutePath` in this line with `.toURI().toString()`.